Source/Core/DolphinTool/ExtractCommand.cpp

```diff
@@ -0,0 +1,309 @@
+// Copyright 2024 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <DolphinTool/ExtractCommand.h>
+#include <OptionParser.h>
+#include <filesystem>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <future>
+#include <iostream>
+#include "DiscIO/DiscExtractor.h"
+#include "DiscIO/DiscUtils.h"
+#include "DiscIO/Filesystem.h"
+#include "DiscIO/Volume.h"
+
+namespace DolphinTool
+{
+static void ExtractFile(const DiscIO::Volume& disc_volume, const DiscIO::Partition& partition,
+                        const std::string& path, const std::string& out)
+{
+  const DiscIO::FileSystem* filesystem = disc_volume.GetFileSystem(partition);
+  if (!filesystem)
+    return;
+
+  ExportFile(disc_volume, partition, filesystem->FindFileInfo(path).get(), out);
+}
+
+static std::unique_ptr<DiscIO::FileInfo> GetFileInfo(const DiscIO::Volume& disc_volume,
+                                                     const DiscIO::Partition& partition,
+                                                     const std::string& path)
+{
+  const DiscIO::FileSystem* filesystem = disc_volume.GetFileSystem(partition);
+  std::unique_ptr<DiscIO::FileInfo> info = filesystem->FindFileInfo(path);
+  return info;
+}
+
+static int VolumeSupported(const DiscIO::Volume& disc_volume)
```

Comment:
EXIT_FAILURE and EXIT_SUCCESS are normally only used as return values from main(). Here you can just return a bool instead.